conjur_api/providers/oidc_authentication_strategy.py

```diff
@@ -29,10 +29,23 @@ async def _send_authenticate_request(self, ssl_verification_data, connection_inf
             'service_id': connection_info.service_id,
             'account': connection_info.conjur_account,
         }
-        data = f"id_token={creds.password}"
 
-        response = await invoke_endpoint(HttpVerb.POST, ConjurEndpoint.AUTHENTICATE_OIDC,
+        if (not creds.code or not creds.code_verifier or not creds.nonce) and (not creds.username or not creds.password):
```

Comment:
Please add a comment explaining the decision tree here for using V1 / V2 of the OIDC authenticator.